### PR TITLE
Jira build 96 add forcepull to buildoverrides

### PR DESCRIPTION
**Pull request by coreydaley**
Use the ForcePull pointer from BuildOverridesConfig

https://issues.redhat.com/browse/BUILD-96

### CODE REVIEW

File: pkg/build/controller/build/overrides/overrides.go [hunk 1]
```diff
@@ -32,22 +32,28 @@ func (b BuildOverrides) ApplyOverrides(pod *corev1.Pod) error {
...
 		}
 		if build.Spec.Strategy.SourceStrategy != nil {
-			klog.V(5).Infof("Setting source strategy ForcePull to true in build %s/%s", build.Namespace, build.Name)
-			build.Spec.Strategy.SourceStrategy.ForcePull = true
+			klog.V(5).Infof("Setting source strategy ForcePull to %t in build %s/%s", *b.Config.ForcePull, build.Namespace, build.Name)
+			build.Spec.Strategy.SourceStrategy.ForcePull = *b.Config.ForcePull
 		}
 		if build.Spec.Strategy.CustomStrategy != nil {
-			err := applyForcePullToPod(pod)
+			pullPolicy := corev1.PullNever
```
Comment (adambkaplan) [inline pkg/build/controller/build/overrides/overrides.go:45]: This should be `corev1.IfNotPresent`.

Comment (coreydaley) [reply pkg/build/controller/build/overrides/overrides.go]: Updated, thanks.

File: pkg/build/controller/build/overrides/overrides.go [hunk 2]
```diff
@@ -32,22 +32,28 @@ func (b BuildOverrides) ApplyOverrides(pod *corev1.Pod) error {
 
 	klog.V(4).Infof("Applying overrides to build %s/%s", build.Namespace, build.Name)
 
-	if b.Config.ForcePull {
+	if b.Config.ForcePull != nil {
```
Comment (gabemontero) [inline pkg/build/controller/build/overrides/overrides.go:35]: could we change this to `if b.Config.ForcePull != nil && *b.Config.ForcePull` and then leave most of what is below the same?

File: pkg/build/controller/build/overrides/overrides.go (hunk 1 above)
Comment (gabemontero) [reply pkg/build/controller/build/overrides/overrides.go:45]: wasn't this resolved incorrectly ... the code change still shows defaulting to Never 

Per https://github.com/kubernetes/api/blob/master/core/v1/types.go#L2243 the default is IfNotPresent anyways

seems like you should leave `applyForcePullToPod` as is and do not add the pullPolicy param

**Comment by gabemontero** [comment]
in case it is masked by github's PR review ... I think the comment from 13 days ago around line 45 of overrides.go was resolved incorrectly @coreydaley @adambkaplan 

the default should be IfNotPresent, which is what k8s go doc says, and hence I don't think `applyForcePullToPod` should need modification

**Comment by coreydaley** [comment]
If the system admin has set ForcePull to false, shouldn't we propagate that down also?
So if a user has set it to corev1.PullAlways, and the sys admin set ForcePull to false, shouldn't that change the users config to corev1.IfNotPresent?

File: pkg/build/controller/build/overrides/overrides.go (hunk 2 above)
Comment (gabemontero) [reply pkg/build/controller/build/overrides/overrides.go:35]: duh, nevermind ... if they switch back to false ... see @coreydaley 's https://github.com/openshift/openshift-controller-manager/pull/123#issuecomment-655558903

feel free to resolve this one @coreydaley

**Comment by gabemontero** [comment]
> If the system admin has set ForcePull to false, shouldn't we propagate that down also?

duh ... correct ... also updated my comment above ... feel free to resolve

> So if a user has set it to corev1.PullAlways, and the sys admin set ForcePull to false, shouldn't that change the users config to corev1.IfNotPresent?

ok for this one, the user does not have access to set pull always, etc. ... they just control force pull, and force pull should either translate to always or ifnotpresent 

Your code still has `pullPolicy := corev1.PullNever`

**Comment by coreydaley** [comment]
> Your code still has `pullPolicy := corev1.PullNever`

Yep, I will fix that, I think it got reset when I made a recent change :(

**Comment by coreydaley** [comment]
@gabemontero changed to IfNotPresent instead of Never

File: pkg/build/controller/build/overrides/overrides_test.go [hunk 3]
```diff
@@ -34,7 +34,8 @@ func TestBuildOverrideForcePull(t *testing.T) {
 	ops := []admission.Operation{admission.Create, admission.Update}
 	for _, test := range tests {
 		for _, op := range ops {
-			overrides := BuildOverrides{Config: &openshiftcontrolplanev1.BuildOverridesConfig{ForcePull: true}}
+			truePtr := true
```
[the remark this answers is not in the document]
Comment (coreydaley) [reply pkg/build/controller/build/overrides/overrides_test.go]: updated and pushed

**Comment by gabemontero** [comment]
LGTM.   Let’s wait and see if @adambkaplan wants to chime in.  If he
doesn’t by scrum tomorrow we’ll ask him then.

**Comment by coreydaley** [comment]
@adambkaplan Do you have any comments on this?

File: pkg/build/controller/build/overrides/overrides.go [hunk 4]
```diff
@@ -98,14 +104,14 @@ func (b BuildOverrides) ApplyOverrides(pod *corev1.Pod) error {
 	return common.SetBuildInPod(pod, build)
 }
 
-func applyForcePullToPod(pod *corev1.Pod) error {
+func applyForcePullToPod(pod *corev1.Pod, pullPolicy corev1.PullPolicy) error {
```
Comment (adambkaplan) [inline pkg/build/controller/build/overrides/overrides.go:107]: nit - can rename this to `applyPullPolicyToPod`

Comment (coreydaley) [reply pkg/build/controller/build/overrides/overrides.go]: updated in current push

**Comment by coreydaley** [comment]
@adambkaplan @gabemontero all comments have been addressed and all requested updates completed. ptal

**Pull request merged**